### PR TITLE
TaskManager refactoring

### PR DESCRIPTION
**Pull request by antonovchtd**
Refactoring: eliminate map::at use, better getters for TaskManager class (as per @vladimir-moiseiev comment)

### CODE REVIEW

File: src/model/TaskManager.cpp [hunk 1]
```diff
@@ -10,6 +10,42 @@ TaskManager::TaskManager() : gen_(std::shared_ptr<IDGenerator>(new IDGenerator))
 TaskManager::TaskManager(const std::shared_ptr<IDGenerator> &generator) : gen_(generator) {
 }
 
+void TaskManager::AddChild(const Core::TaskID &parent, const Core::TaskID &child) {
```
Comment (vladimir-moiseiev) [inline src/model/TaskManager.cpp]: `parent_id``child_id`

File: src/model/TaskManager.cpp [hunk 2]
```diff
@@ -48,76 +84,85 @@ std::vector<Core::TaskEntity> TaskManager::getTasks() const {
...
     for (const auto &kv : tasks_) {
-        if (kv.second.first.label() == label) {
-            Core::TaskEntity te;
-            te.set_allocated_id(new Core::TaskID(kv.first));
-            te.set_allocated_data(new Core::Task(kv.second.first));
-            tasks.push_back(te);
+        for (const auto &task_label : kv.second.first.labels()) {
+            if (task_label == label) {
+                Core::TaskEntity te;
+                te.set_allocated_id(new Core::TaskID(kv.first));
```
Comment (vladimir-moiseiev) [inline src/model/TaskManager.cpp]: I'm pretty sure you can avoid `set_allocated*` syntax, and `new`

File: src/model/TaskManager.cpp [hunk 3]
```diff
@@ -48,76 +84,85 @@ std::vector<Core::TaskEntity> TaskManager::getTasks() const {
...
+                te.set_allocated_data(new Core::Task(kv.second.first));
+                tasks.push_back(te);
```
Comment (vladimir-moiseiev) [inline src/model/TaskManager.cpp]: consider `emplace_back`

File: src/model/TaskManager.cpp [hunk 4]
```diff
@@ -48,76 +84,85 @@ std::vector<Core::TaskEntity> TaskManager::getTasks() const {
...
     std::vector<Core::TaskEntity> tasks;
     for (const auto &kv : tasks_) {
-        if (kv.second.first.label() == label) {
-            Core::TaskEntity te;
-            te.set_allocated_id(new Core::TaskID(kv.first));
-            te.set_allocated_data(new Core::Task(kv.second.first));
-            tasks.push_back(te);
+        for (const auto &task_label : kv.second.first.labels()) {
+            if (task_label == label) {
+                Core::TaskEntity te;
```
Comment (vladimir-moiseiev) [inline src/model/TaskManager.cpp]: consider factory method or builder for creating `TaskEntity`

File: src/model/TaskManager.cpp [hunk 5]
```diff
@@ -48,76 +84,85 @@ std::vector<Core::TaskEntity> TaskManager::getTasks() const {
...
         }
         tasks.insert(tasks.end(), ch_tasks.begin(), ch_tasks.end());
     }
     return tasks;
 }
 
 ActionResult TaskManager::Delete(const Core::TaskID &id, bool deleteChildren) {
-    try {
-        if (!tasks_.at(id).second.children().empty() && !deleteChildren)
+    if (Validate(id)) {
```
Comment (vladimir-moiseiev) [inline src/model/TaskManager.cpp]: you don't "Validate" `id`, you're checking if it exists

File: src/model/TaskManager.h [hunk 6]
```diff
@@ -33,10 +33,21 @@ class TaskManager : public ModelInterface {
     ActionResult Delete(const Core::TaskID &id, bool deleteChildren) override;
     ActionResult Validate(const Core::TaskID &id) const override;
     ActionResult AddLabel(const Core::TaskID &, const std::string &) override;
+    ActionResult ClearLabel(const Core::TaskID &, const std::string &) override;
```
Comment (vladimir-moiseiev) [inline src/model/TaskManager.h:37]: this branch should not me from `main`. you have these changes already in other review

**Review by vladimir-moiseiev — APPROVED** [review]

File: src/model/TaskManager.cpp (hunk 1 above)
Comment (antonovchtd) [reply src/model/TaskManager.cpp]: renamed

File: src/model/TaskManager.cpp (hunk 2 above)
Comment (antonovchtd) [reply src/model/TaskManager.cpp]: I could make_unique + release() ?

Comment (antonovchtd) [reply src/model/TaskManager.cpp]: I'll use `te.mutable_id()->CopyFrom(kv.first);`

File: src/model/TaskManager.cpp (hunk 3 above)
Comment (antonovchtd) [reply src/model/TaskManager.cpp]: Will do for the future. Here, TaskEntity does not have a id+data constructor (and I want the parent field to stay unassigned)

Comment (antonovchtd) [reply src/model/TaskManager.cpp]: ultimately, used `emplace_back` with copy constructor and factory method

File: src/model/TaskManager.cpp (hunk 4 above)
Comment (antonovchtd) [reply src/model/TaskManager.cpp]: implemented

File: src/model/TaskManager.cpp (hunk 5 above)
Comment (antonovchtd) [reply src/model/TaskManager.cpp]: renamed to `IsPresent`

**Pull request merged**